fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java

```diff
@@ -252,10 +271,19 @@ public Cost visitPhysicalHashJoin(
                     leftRowCount + rightRowCount,
                     penalty);
         }
+
+        if (context.isBroadcastJoin()) {
+            double broadcastJoinPenalty = broadCastJoinBalancePenalty(probeStats, buildStats);
+            return CostV1.of(leftRowCount * broadcastJoinPenalty + rightRowCount + outputRowCount,
```

Comment:
I think we would better to set up cost system on more dimension, for distinguishing the bc and shuffle, for example, we would consider exchange data cost, parallelism info to compare these two, by simulating the real execution difference between bc and shuffle as close as possible. In comparison, the introduced penalty parameter is meaningless, such as BROADCAST_JOIN_SKEW_RATIO, and will easily to be broken under different scenario in the future.

Reply:
this idea can be implemented in CostModelV2. In V1, we do not have this dimension